plugin/src/main/java/at/helpch/chatchat/util/Kyorifier.java

```diff
@@ -0,0 +1,71 @@
+package at.helpch.chatchat.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class Kyorifier {
+    private static final Map<Character, String> COLOURS = new HashMap<>(15);
+    private static final Map<Character, String> FORMATTERS = new HashMap<>(6);
+    private static final Pattern pattern = Pattern.compile("&(?<code>[\\da-fk-or])|[&{\\[<]?[#x](?<hex>(&?[A-Fa-f\\d]){6})[}\\]>]?");
+
+    static {
+        COLOURS.put('0', "black");
+        COLOURS.put('1', "dark_blue");
+        COLOURS.put('2', "dark_green");
+        COLOURS.put('3', "dark_aqua");
+        COLOURS.put('4', "dark_red");
+        COLOURS.put('5', "dark_purple");
+        COLOURS.put('6', "gold");
+        COLOURS.put('7', "gray");
+        COLOURS.put('8', "dark_gray");
+        COLOURS.put('9', "blue");
+        COLOURS.put('a', "green");
+        COLOURS.put('b', "aqua");
+        COLOURS.put('c', "red");
+        COLOURS.put('d', "light_purple");
+        COLOURS.put('e', "yellow");
+        COLOURS.put('f', "white");
```

Comment:
I think we can replace this with [`NamedTextColor.COLOR.toString()`](https://github.com/KyoriPowered/adventure/blob/d32ab6b28f7ff43b0d1c89ca2a886c2a34b2d696/api/src/main/java/net/kyori/adventure/text/format/NamedTextColor.java#L296-L298). [NamedTextColor](https://github.com/KyoriPowered/adventure/blob/d32ab6b28f7ff43b0d1c89ca2a886c2a34b2d696/api/src/main/java/net/kyori/adventure/text/format/NamedTextColor.java#L279) accepts a `name` parameter which is returned by `#toString()`. Is unlikely that they will ever change the name of these colors, but I think is a better not to hardcode the values.